### DISCUSSION TITLE
What is the proper way to get CSS aliases working in .css files?

### DISCUSSION THREAD

**Discussion by robahtou** [vercel/next.js#71982] [Help]
### Summary

I recently updated to v15 stable and set the `--turbo` flag for development. And it worked great with minimal code updates, except for the CSS.

I'm getting a bunch of CSS errors, and all of them are associated with how CSS aliases were being used in the code.

For example, 

This worked prior to the upgrade in dev & build: 
`background-image: url('~@Images/mobile-calculate-people.png');`

Now with the `--turbo` flag on dev it breaks but the build works.

If I update the url alias to `background-image: url('@Images/mobile-calculate-people.png');`, dev works but the build breaks.
And no combination of alias is working for both dev and build. So I have to resort to relative paths names. 

I can't tell if this is a bug or not. 
Does anyone know how to properly configure CSS aliases? 

I tried following the following references but still no avail.
* https://stackoverflow.com/questions/72179248/how-do-i-configure-nextjs-to-correctly-handle-alias-paths-within-scss
* https://stackoverflow.com/questions/76607491/css-import-aliases-with-next-js

I tried setting things up using the `turbo` key but it doesn't work either:
* https://nextjs.org/docs/app/api-reference/next-config-js/turbo

```experimental: {
    turbo: {
      resolveAlias: {
        '@Images': './src/assets/images'
      }
    }
  }
```
Is this a bug, configuration issue, or perhaps a css syntax thing?

N.B.  I can do this at the top of my css files `@import url('@Styles/my_styles.css');` and works both in dev & build. But when I do it inline like in the `background-image`, it fails

**Comment by aniruddhaadak80** [comment]
Hi @robahtou ,

Solutions to get CSS aliases working in your Next.js project after upgrading to version 15 with the `--turbo` flag:

1. **Update `next.config.js`**:
   Ensure your CSS alias is correctly configured:
   ```javascript
   const path = require('path');

   module.exports = {
     turbo: {
       resolveAlias: {
         '@Images': path.resolve(__dirname, './src/assets/images'),
       },
     },
   };
   ```

2. **Use Correct URL Syntax**:
   In your CSS files, use the alias with the correct syntax:
   ```css
   background-image: url('@Images/mobile-calculate-people.png');
   ```

3. **Restart the Development Server**:
   After making changes to `next.config.js`, restart your development server to apply the new configuration. 

If issues persist, consider using relative paths as a fallback.

Thanks.

**Reply by robahtou** [reply]
This does not resolve the build breaking. Dev works fine but when running the build it breaks.

**Comment by sekiranat** [comment]
Yes, I see the same problem with scss
My config:

`const nextConfig: NextConfig = {
  experimental: {
    turbo: {
      resolveAlias: {
        '@Assets': path.resolve(__dirname, './src/app/assets'),
      },
    },
  },
  sassOptions: {
    additionalData: `@import "@/app/assets/scss/variables.scss";`,
    sassOptions: {
      includePaths: [path.join(__dirname, 'src', 'app')],
    },
  },
}`

scss: background: url('@Assets/icons/search.svg') no-repeat;

Relative paths work fine
